Seminars/Sem.02/Pract.02/Task7/Task7.cpp

```diff
@@ -0,0 +1,238 @@
+﻿#include <iostream>
+#include <fstream>
+
+const int MAX_LENGTH = 128;
+const int MAX_NUM_OF_USERS = 100;
+const int MAX_BUFF_SIZE = 1024;
+
+struct User
+{
+    char name[MAX_LENGTH];
+    char email[MAX_LENGTH];
+    char password[MAX_LENGTH];
+};
+
+struct System
+{
+    User users[MAX_NUM_OF_USERS];
+};
+
+void toBeginningOfFile()
+{
+    std::ifstream usersFile("users_system.txt", std::ios::app | std::ios::_Nocreate);
+
+    if (!usersFile.is_open())
+    {
+        std::cout << "Error ocurred when trying to open the file." << std::endl;
+
+        return;
+    }
+
+    usersFile.clear();
+    usersFile.seekg(0, std::ios::beg);
+}
+
+void writeUserToFile(const User& user)
+{
+    std::ofstream usersFile("users_system.txt", std::ios::app | std::ios::_Nocreate);
```

Comment:
with this combination of ios::app and ios::nocreate you always create files even if the file does not exists If you want to use nocreate you will hit errors when trying to register the first user. But ios::app is enough in this situation. It creates the file if the file does not exist and opens the file (if it exists) without deleting information and writes information at the end